python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -303,43 +308,35 @@ def get_results(
         self,
         *,
         implicit_materializations: bool = True,
-    ) -> Sequence[PipesExecutionResult]:
+    ) -> Iterator[PipesExecutionResult]:
         """:py:class:`PipesExecutionResult` objects reported from the external process.
 
+        When this is called it clears the results buffer.
+
         Args:
             implicit_materializations (bool): Create MaterializeResults for expected assets
                 even was nothing is reported from the external process.
 
         Returns:
             Sequence[PipesExecutionResult]: Result reported by external process.
         """
-        reported = self.message_handler.get_reported_results()
+        reported = self.message_handler.clear_result_queue()
         if not implicit_materializations:
-            return reported
+            yield from reported
+            return
 
-        reported_keys = set(
-            result.asset_key for result in reported if isinstance(result, MaterializeResult)
-        )
-        implicit = (
+        reported_keys = set()
+        for result in reported:
+            if isinstance(result, MaterializeResult):
+                reported_keys.add(result.asset_key)
+            yield result
+
+        yield from (
             MaterializeResult(asset_key=key)
             for key in self.context.selected_asset_keys
             if key not in reported_keys
         )
 
-        return (
-            *reported,
-            *implicit,
-        )
-
-    @public
-    def get_reported_results(self) -> Sequence[PipesExecutionResult]:
```

Comment:
accidental addition from iterating, missed getting rid of this in final land of #17921